lib/benchee/system.ex

```diff
@@ -32,4 +34,52 @@ defmodule Benchee.System do
     end
   end
 
+  def num_cores do
+    System.schedulers_online()
+  end
+
+  def os do
+    {_, name} = :os.type()
+    os(name)
+  end
+
+  defp os(:darwin), do: :macOS
+  defp os(:nt), do: :Windows
+  defp os(_), do: :Linux
+
+  def cpu_speed do
+    cpu_speed(os())
+  end
+
+  defp cpu_speed(:macOS) do
+    {output, exit_code} = System.cmd("sysctl", ["-n", "machdep.cpu.brand_string"])
+    if exit_code > 0 do
+      IO.puts("Something went wrong trying to get system information")
+    end
+    output
+  end
+  defp cpu_speed(:Windows) do
+    "N/A"
```

Comment:
ah yeah totally forgot to comment on that, feel free to do that in this PR or in separate per OS or per thing (cpu speed/memory) PRs :)